### PR TITLE
Enable_gptj

### PR DESCRIPTION
**Pull request by ZhaiFeiyue**
# What does this PR do?

as title

FP32 outputs:
image

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by ZhaiFeiyue** [comment]
@regisss gptj is here, have a look if you are free

File: optimum/habana/transformers/models/gptj/modeling_gptj.py [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+
+        if token_idx is not None:
+            # HPU bug WA
+            past_key.index_add_(2, token_idx - 1, key - torch.index_select(past_key, 2, token_idx - 1))
+            past_value.index_add_(2, token_idx - 1, value - torch.index_select(past_value, 2, token_idx - 1))
+            key = past_key
+            value = past_value
+        else:
+            key = torch.cat([past_key, key], dim=2)
+            value = torch.cat([past_value, value], dim=2)
```
Comment (regisss) [inline optimum/habana/transformers/models/gptj/modeling_gptj.py:75]:
```suggestion
            key = torch.cat([past_key, key], dim=-2)
            value = torch.cat([past_value, value], dim=-2)
```
Not sure this changes anything, but it's just to stay as close as possible to the Transformers implementation.
Maybe this has an impact on lines 69 and 70 above?

File: optimum/habana/transformers/models/gptj/modeling_gptj.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+        attention_mask = kwargs.get("attention_mask", None)
+        position_ids = kwargs.get("position_ids", None)
+
+        if attention_mask is not None and position_ids is None:
+            # create position_ids on the fly for batch generation
+            position_ids = attention_mask.long().cumsum(-1) - 1
+            position_ids.masked_fill_(attention_mask == 0, 1)
+            if past_key_values:
+                if past_key_values:
```
Comment (regisss) [inline optimum/habana/transformers/models/gptj/modeling_gptj.py:355]: Same as the line right above, to remove?

**Review by regisss — COMMENTED** [review]
@ZhaiFeiyue I just left a couple of minor comments and after they are addressed we will merge!

**Comment by regisss** [comment]
@ZhaiFeiyue FYI, I tried bf16 and DeepSpeed-inference and it's the same as OPT: bf16 worked well and same error with DS.

File: optimum/habana/transformers/models/gptj/modeling_gptj.py (hunk 1 above)
Comment (ZhaiFeiyue) [reply optimum/habana/transformers/models/gptj/modeling_gptj.py]: Yes, we should stay as close the the Transformers implementation. Rank of key and value is 4, so -2 is same with 2.

File: optimum/habana/transformers/models/gptj/modeling_gptj.py (hunk 2 above)
Comment (ZhaiFeiyue) [reply optimum/habana/transformers/models/gptj/modeling_gptj.py]: done

**Review by regisss — APPROVED** [review]
LGTM!

**Pull request merged**